Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceStateMachine+State.swift

```diff
@@ -102,6 +124,9 @@ extension ExperienceStateMachine.State: Equatable {
             return experience1.id == experience2.id && stepIndex1 == stepIndex2 && markComplete1 == markComplete2
         case let (.endingExperience(experience1, stepIndex1, markComplete1), .endingExperience(experience2, stepIndex2, markComplete2)):
             return experience1.id == experience2.id && stepIndex1 == stepIndex2 && markComplete1 == markComplete2
+        case let (.failing(targetState1, _), .failing(targetState2, _)):
+            // TODO: to fully compare equality, would need to make SideEffect equatable (many cascading updates)
+            return targetState1 == targetState2
```

Comment:
Note the TODO - I believe this is only really used in unit tests to compare states. This one is tricky, it would require making SideEffect equatable, then making Action equatable, and perhaps several other pieces of the associated values in those enums.